vpr/src/route/gen/rr_graph_uxsdcxx.h

```diff
@@ -23,6 +23,7 @@
 #include <stddef.h>
 #include <stdint.h>
 #include "pugixml.hpp"
+#include "globals.h"
```

Comment:
Actually, introducing global variable is a critical concern to me. It makes the code hard to maintain because it is hard to know what are the required data structure for each function.

This motivates me to review what the ``rr_graph_uxsdcxx.h`` is doing. It actually creates its own context in this file, being independent from the RRGraph and other VPR data structure. This looks clean to me. It makes sense that it should have it own ``node_capacity()`` function so that they are isolated from RRGraph data structure. 

https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/6e442d112d843f310f5f6bfd88663186981a036f/vpr/src/route/rr_graph_uxsdcxx_serializer.h#L803-L805

So, I suggest not to introduce the global variables to this file. But we should update the API in the ``rr_graph_uxsdcxx_serializer.h``. Actually, I found how they access the rr_graph inside. We can update the ``get_node_capacity()`` like this:

https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/6e442d112d843f310f5f6bfd88663186981a036f/vpr/src/route/rr_graph_uxsdcxx_serializer.h#L809-L812

Reply:
@vaughnbetz I found something in the RRGraph serializer which may conflicts with your previous comments. See if my suggestion makes sense or not.

Reply:
OK, I agree with you @tangxifan .  I missed that they weren't accessing the same rr_graph data structure, but were instead accessing their own local version.  In that case, I agree my earlier comments are wrong and we should revert to what Ethan had (which I think is basically the same code as what you propose above).

Sorry for the churn @ethanroj23

Reply:
No worries! I have now replaced the code, with the commented out sections of code still removed, so we should be good to go! @vaughnbetz let me know if there is anything else that should be altered here. Thanks!

Reply:
@vaughnbetz Thanks for double-check!

@ethanroj23 I have no concerns on the PR. Look good to me.